src/app/content/hooks/locationChange/resolveContent.spec.ts

```diff
@@ -105,4 +117,25 @@ describe('locationChange', () => {
     await hook(helpers, match);
     expect(helpers.archiveLoader.mock.loadBook).toHaveBeenCalledWith('testbook1-uuid', undefined);
   });
+
+  it('uses uuid if present', async() => {
+    const versionedSlugParams = {
+      page: match.params.page,
+      uuid: 'longidin-vali-dfor-mat1-111111111111',
+      version: '1.0',
+    } as Params;
+
+    match.params = versionedSlugParams;
+    helpers.archiveLoader.mockBook({
+      ...book,
+      version: undefined as any as string,
+    });
+    helpers.archiveLoader.mockPage({
+      ...book,
+      version: undefined as any as string,
+    }, page, 'test-page-1');
```

Comment:
i don't think these calls to `mockBook` and `mockPage` are currently doing anything, because you're creating them based on the `book` but you're not querying that book. if you use that book variable as your base, but do 

```
helpers.archiveLoader.mockBook({
      ...book,
      id: 'longidin-vali-dfor-mat1-111111111111'
      version: undefined as any as string,
    });
```

then you will not need to add the whole json file